components/script/script_module.rs

```diff
@@ -337,6 +332,12 @@ pub fn fetch_single_module_script(
         return;
     }
 
+    // Step 4.
+    {
+        let global = owner.global();
+        global.set_module_map(url.clone(), ModuleObject::Fetching);
```

Comment:
@jdm As we discussed, `fetching` should be set here that it should avoid duplicated fetching. But, this will get `BorrowMutError` while running the `the-script-element/module/single-evaluation-1.html` test case 🤔

<details>
<summary>CRASH output in my local</summary>

```
pid:67314 DomRefCell<T> already borrowed: BorrowMutError (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }, at src/libcore/result.rs:999)
 0:21.64 pid:67314 stack backtrace:
 0:21.64 pid:67314    0: __ZN9backtrace9backtrace20trace_unsynchronized17hb6cc618e5dfe44c1E (0x1167877bc)
 0:21.64 pid:67314              at /Users/cybai/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.26/src/backtrace/mod.rs:66
 0:21.64 pid:67314    1: __ZN9backtrace9backtrace5trace17h01201da60eac6e66E (0x116787742)
 0:21.64 pid:67314              at /Users/cybai/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.26/src/backtrace/mod.rs:53
 0:21.64 pid:67314    2: __ZN9backtrace7capture9Backtrace6create17hf9586ac2b3b05b79E (0x116777266)
 0:21.64 pid:67314              at /Users/cybai/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.26/src/capture.rs:163
 0:21.64 pid:67314    3: __ZN9backtrace7capture9Backtrace3new17h4c2e2bf10cc5a82fE (0x1167771a4)
 0:21.64 pid:67314              at /Users/cybai/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.26/src/capture.rs:126
 0:21.64 pid:67314    4: __ZN5servo4main28_$u7b$$u7b$closure$u7d$$u7d$17hccd339d13c20a4aeE (0x10f124194)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/ports/glutin/main2.rs:119
 0:21.64 pid:67314    5: __ZN3std9panicking20rust_panic_with_hook17h7c45ae722ecd7c4eE (0x11707ec40)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panicking.rs:479
 0:21.64 pid:67314    6: __ZN3std9panicking18continue_panic_fmt17hdbfce1688aed9892E (0x11707e68c)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panicking.rs:382
 0:21.64 pid:67314    7: _rust_begin_unwind (0x11707e578)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panicking.rs:309
 0:21.64 pid:67314    8: __ZN4core9panicking9panic_fmt17he225f000d3fdc81fE (0x11709fba1)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libcore/panicking.rs:85
 0:21.64 pid:67314    9: __ZN4core6result13unwrap_failed17h1303ee6a9e0b5566E (0x11657ef32)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libcore/macros.rs:18
 0:21.64 pid:67314   10: __ZN4core6result19Result$LT$T$C$E$GT$6expect17hf91a0e883f30c7a9E (0x10fe5cc9e)
 0:21.64 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libcore/result.rs:827
 0:21.64 pid:67314   11: __ZN6script3dom8bindings4cell19DomRefCell$LT$T$GT$10borrow_mut17ha39d0fcd7c412f6fE (0x11041e6a8)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/bindings/cell.rs:86
 0:21.64 pid:67314   12: __ZN6script3dom11globalscope11GlobalScope14set_module_map17hf7008fe37341a7d8E (0x110047ded)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/globalscope.rs:321
 0:21.64 pid:67314   13: __ZN6script13script_module26fetch_single_module_script17h3a9698d3c1507a74E (0x112289ec8)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_module.rs:338
 0:21.64 pid:67314   14: __ZN6script13script_module20create_module_script17hb1d05dfc17fea2f3E (0x112289b30)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_module.rs:294
 0:21.64 pid:67314   15: __ZN6script13script_module25fetch_module_script_graph17hb5d2d258f5dede59E (0x11228992e)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_module.rs:261
 0:21.64 pid:67314   16: __ZN6script3dom17htmlscriptelement17HTMLScriptElement7prepare17h588b8884e139d7e2E (0x1114379a1)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/htmlscriptelement.rs:544
 0:21.64 pid:67314   17: __ZN6script3dom11servoparser11ServoParser8tokenize17h41dcf74713022f5dE (0x1113e6bc2)
 0:21.64 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/servoparser/mod.rs:523
 0:21.64 pid:67314   18: __ZN6script3dom11servoparser11ServoParser13do_parse_sync17hb9e2d0fd326b3ea1E (0x111c9ffb2)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/servoparser/mod.rs:477
 0:21.65 pid:67314   19: __ZN6script3dom11servoparser11ServoParser10parse_sync28_$u7b$$u7b$closure$u7d$$u7d$17h883749352538db15E (0x1113e64d7)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/servoparser/mod.rs:459
 0:21.65 pid:67314   20: __ZN14profile_traits4time7profile17hd66dee34a2e797b6E (0x11081c2a4)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/profile_traits/time.rs:146
 0:21.65 pid:67314   21: __ZN6script3dom11servoparser11ServoParser10parse_sync17h7b04249d5c0accb7E (0x111c9fc62)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/servoparser/mod.rs:451
 0:21.65 pid:67314   22: __ZN6script3dom11servoparser11ServoParser43resume_with_pending_parsing_blocking_script17h39a16c714434731dE (0x111c9e16a)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/servoparser/mod.rs:296
 0:21.65 pid:67314   23: __ZN6script3dom8document8Document39process_pending_parsing_blocking_script17h03300ad7c392ee94E (0x1119bc0b2)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/document.rs:2095
 0:21.65 pid:67314   24: __ZN6script3dom8document8Document38pending_parsing_blocking_script_loaded17hf75b8913ae136f29E (0x1119bbd59)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/document.rs:2081
 0:21.65 pid:67314   25: __ZN100_$LT$script..dom..htmlscriptelement..ClassicContext$u20$as$u20$net_traits..FetchResponseListener$GT$20process_response_eof17h0cee41ea016f089aE (0x11143528a)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/dom/htmlscriptelement.rs:264
 0:21.65 pid:67314   26: __ZN76_$LT$net_traits..FetchResponseMsg$u20$as$u20$net_traits..Action$LT$T$GT$$GT$7process17h32b9faf6ed201015E (0x1117f9c74)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/net_traits/lib.rs:260
 0:21.65 pid:67314   27: __ZN101_$LT$script..network_listener..ListenerTask$LT$A$C$Listener$GT$$u20$as$u20$script..task..TaskOnce$GT$8run_once17hb58fb791bdfd27fbE (0x1121e71c0)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/network_listener.rs:115
 0:21.65 pid:67314   28: __ZN81_$LT$script..task..CancellableTask$LT$T$GT$$u20$as$u20$script..task..TaskOnce$GT$8run_once17hf0950ae21c2a9987E (0x11001b42f)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/task.rs:122
 0:21.65 pid:67314   29: __ZN43_$LT$T$u20$as$u20$script..task..TaskBox$GT$7run_box17hc1761132d1bcdfadE (0x11000fcc3)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/task.rs:64
 0:21.65 pid:67314   30: __ZN6script13script_thread12ScriptThread22handle_msg_from_script17hd4775841ef1cb591E (0x11017554b)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:1822
 0:21.65 pid:67314   31: __ZN6script13script_thread12ScriptThread11handle_msgs28_$u7b$$u7b$closure$u7d$$u7d$17h89f75e9d322dfadaE (0x11084ca4f)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:1423
 0:21.65 pid:67314   32: __ZN6script13script_thread12ScriptThread13profile_event17h2ccf84c91ebb28afE (0x11084e026)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:1659
 0:21.65 pid:67314   33: __ZN6script13script_thread12ScriptThread11handle_msgs17hd36e75d59af4acf3E (0x11017131f)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:1416
 0:21.65 pid:67314   34: __ZN6script13script_thread12ScriptThread5start17h77bdb9989d0cfea4E (0x11016ec5c)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:1249
 0:21.65 pid:67314   35: __ZN90_$LT$script..script_thread..ScriptThread$u20$as$u20$script_traits..ScriptThreadFactory$GT$6create28_$u7b$$u7b$closure$u7d$$u7d$28_$u7b$$u7b$closure$u7d$$u7d$17hadf7ca7b15a7aa78E (0x110849754)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:785
 0:21.65 pid:67314   36: __ZN14profile_traits3mem12ProfilerChan25run_with_memory_reporting17h7539f8cad1c91a5eE (0x110faa857)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/profile_traits/mem.rs:88
 0:21.65 pid:67314   37: __ZN90_$LT$script..script_thread..ScriptThread$u20$as$u20$script_traits..ScriptThreadFactory$GT$6create28_$u7b$$u7b$closure$u7d$$u7d$17h08366be034cf9b6fE (0x110849e89)
 0:21.65 pid:67314              at /Users/cybai/codespace/mozilla/servo/components/script/script_thread.rs:783
 0:21.65 pid:67314   38: __ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hbf8688160175f8e8E (0x1111fbc92)
 0:21.65 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/sys_common/backtrace.rs:77
 0:21.65 pid:67314   39: __ZN3std6thread7Builder15spawn_unchecked28_$u7b$$u7b$closure$u7d$$u7d$28_$u7b$$u7b$closure$u7d$$u7d$17ha3af6f879985ec30E (0x110a7a552)
 0:21.65 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/thread/mod.rs:470
 0:21.65 pid:67314   40: __ZN101_$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$$LP$$RP$$GT$$GT$9call_once17he0d51649b84d208cE (0x1106defe2)
 0:21.65 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panic.rs:309
 0:21.65 pid:67314   41: __ZN3std9panicking3try7do_call17h0b8c80dfe7aebd7bE (0x112128d52)
 0:21.65 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panicking.rs:294
 0:21.65 pid:67314   42: ___rust_maybe_catch_panic (0x11708735e)
 0:21.65 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libpanic_unwind/lib.rs:82
 0:21.65 pid:67314   43: __ZN3std9panicking3try17h0f7008330a2b74aeE (0x111f277e4)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panicking.rs:273
 0:21.66 pid:67314   44: __ZN3std5panic12catch_unwind17h19cb9891ad149972E (0x1106f1802)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/panic.rs:388
 0:21.66 pid:67314   45: __ZN3std6thread7Builder15spawn_unchecked28_$u7b$$u7b$closure$u7d$$u7d$17h87278523c30837d1E (0x110a78e3b)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libstd/thread/mod.rs:469
 0:21.66 pid:67314   46: __ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h0656ba8e506c6feeE (0x110885974)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/libcore/ops/function.rs:231
 0:21.66 pid:67314   47: __ZN83_$LT$alloc..boxed..Box$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$9call_once17he9741b9c5e7eb28cE (0x11706e62d)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/liballoc/boxed.rs:746
 0:21.66 pid:67314   48: __ZN3std3sys4unix6thread6Thread3new12thread_start17h3570aea1d6681281E (0x11708687d)
 0:21.66 pid:67314              at /rustc/04a3dd8a872633ca1e4c217d11f741cc35cb19a5/src/liballoc/boxed.rs:746
 0:21.66 pid:67314   49: __pthread_body (0x7fff698152ea)
 0:21.66 pid:67314   50: __pthread_start (0x7fff69818248)
 0:21.66 pid:67314 [2019-07-02T11:03:21Z ERROR servo] DomRefCell<T> already borrowed: BorrowMutError
 0:21.75 pid:67314 Pipeline failed in hard-fail mode.  Crashing!
 0:21.75 pid:67314 thread panicked while processing panic. aborting.
 0:21.90 TEST_END: CRASH, expected OK
```

</details>